### PR TITLE
Pass `TopicResult` rather than `SelectedTopics`

### PR DESCRIPTION
**Pull request by joecowton1**
## What does this change?

Passes the entire `TopicResult` to `DotcomRenderingModel` and creates a `selectedTopics` val there.

Changes the `SelectedTopics` type to be `Option[Seq[Topic]]` rather than `Option[String]]`

### CODE REVIEW

**Review by marjisound — APPROVED** [review]
Looks great

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 1]
```diff
@@ -423,6 +426,8 @@ object DotcomRenderingDataModel {
 
     val matchData = makeMatchData(page)
 
+    val selectedTopics = topicResult.map(x => Seq(Topic(x.`type`, x.name)))
+
```
Comment (mxdvl) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:430]: If you’re not giving it a meaningful name, you can always use the underscore!

```suggestion
    val selectedTopics = topicResult.map(_ => Seq(Topic(_.`type`, _.name)))

```

**Review by mxdvl — APPROVED** [review]

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala (hunk 1 above)
Comment (AshCorr) [reply common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:430]:
```suggestion
    val selectedTopics = topicResult.map(Seq(Topic(_.`type`, _.name)))
```

even

**Review by AshCorr — APPROVED** [review]

File: common/app/model/dotcomrendering/DotcomRenderingDataModel.scala [hunk 2]
```diff
@@ -423,6 +426,7 @@ object DotcomRenderingDataModel {
 
     val matchData = makeMatchData(page)
 
+    val selectedTopics = topicResult.map(Seq(Topic(_.`type`, _.name)))
```
Comment (abeddow91) [inline common/app/model/dotcomrendering/DotcomRenderingDataModel.scala:429]:
```suggestion
    val selectedTopics = topicResult.map(_ => Seq(Topic(_.`type`, _.name)))
```

**Pull request merged**